### DISCUSSION TITLE
Smartling.com for translations instead of Crowdin

### DISCUSSION THREAD

**Discussion by santogu** [facebook/docusaurus#3331] [General]
Hi Docusaurus community,

Our organisation uses https://www.smartling.com/ for translation services of our customer platform. We now have a documentation site on Docusaurus 1.0 and I would love to go with Crowdin but have to evaluate now a Smartling integration.

It would be very nice if anyone could comment on how feasible it is to integrate any other translation service other than Crowdin, and whether anyone happens to have experience with just this particular integration?

Is it crazy to go this path or somewhat feasible? FYI, we have a large group of engineers but I'd prefer not to use them if I don't have to. Thanks for any comments. Cheers.

**Comment by slorber** [comment]
Hi,

We won't invest much time in v1 anymore and not sure it's possible to easily integrate a separate translation system in v1, but it's one of the goal of v2 to allow this.

v2 will move out of alpha as soon as I get i18n working. 
There's an RFC and a draft PR already. https://github.com/facebook/docusaurus/issues/3317
If you want to be an early adopter and explore how to support smartling, we can work together on this.
The plan is to just write some translated md/json files somewhere on the FS and it should work, so a Smartling integration would be a bunch of scripts to download translated content and write it to disk in the correct form.

**Reply by santogu** [reply]
Hi @slorber, thank you so much for your response.

Are you suggesting the goal for v2 in terms of localisation is to be more flexible about vendors to "easily" integrate, unlike v1 where Crowdin is the only viable option?

Or would you just pick Smartling as the guinea pig if we collaborate? :) 

How much effort do you think is required to make this work (I know it's a tricky question)? Depending on effort, I might be able to connect you with our engineers.

**Reply by slorber** [reply]
> Are you suggesting the goal for v2 in terms of localisation is to be more flexible about vendors to "easily" integrate, unlike v1 where Crowdin is the only viable option?

Yes, that's the idea. Also want to support git-based workflows like the one used on ReactJS website. More details in the RFC.

> Or would you just pick Smartling as the guinea pig if we collaborate? :)

We'll provide first class support for Crowdin to help migrate v1 users, but incite the community to build/maintain their own integrations with other SaaS tools like Smartling. I don't think we have the bandwidth to manage all these integrations as a core Docusaurus feature.

> How much effort do you think is required to make this work (I know it's a tricky question)? Depending on effort, I might be able to connect you with our engineers.

I don't know anything about Smartling. I can only tell you how much effort it was to integrate with Crowdin in the upcoming weeks.

**Reply by santogu** [reply]
Thanks for your help. Much appreciated.
We don't have much bandwidth at the moment to support this but I might get back if we prioritise this in the near future. I am very much looking forward to the first v2 stable release!

**Reply by slorber** [reply]
Hi @santogu 

FYI the i18n is merged and doc is written, if you need: https://deploy-preview-4014--docusaurus-2.netlify.app/classic/docs/next/i18n/introduction

**Comment by santogu** [comment]
Thanks for the mention. Happy New Year.

On Sat, Jan 16, 2021 at 3:53 AM Sébastien Lorber <notifications@github.com>
wrote:

> Hi @santogu <https://github.com/santogu>
>
> FYI the i18n is merged and doc is written, if you need:
> https://deploy-preview-4014--docusaurus-2.netlify.app/classic/docs/next/i18n/introduction
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/docusaurus/discussions/3331#discussioncomment-284966>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APSP3D7QJMXYY4PZSFDZ5G3S2BXHDANCNFSM4QKQGSMQ>
> .
>


-- 
*Guido Santo*
Head of Product (Security & Programs)
[image: Rokt secures spot on Crain’s list of fastest-growing companies in
NYC]
<https://rokt.com/resources_post/rokt-secures-spot-on-crains-list-of-fastest-growing-companies-in-nyc>
*M.* +61 (410) 100-754
*A.* L13, 300 Elizabeth Street, Surry Hills, NSW, 2010
<https://goo.gl/maps/FYzqG8488Dp>
USA | Australia | UK | Germany | Singapore | NZ | Canada | Japan
France | Netherlands | Ireland | Spain | Norway | Denmark | Finland | Sweden
